### DISCUSSION TITLE
G7 Sensor failed - works with receiver

### DISCUSSION THREAD

**Discussion by th122** [NightscoutFoundation/xDrip#3390] [Q&A]
Phone: Samsung Galaxy A20e, SM-A202F/DS
Android 11, Security patches 2023-05-01, battery optimization disabled. Prior G6 Sensors and one G7 were running fine. 
Dexcom G7 Rev 05
xDrip nightly e5113d9-2024.03.10
AAPS 3.2.0.3

Update: A new sensor seems to be running smoothly.  Why the other one kept giving an error to xDrip for all its 10 days remains a mystery as of yet.

Problem description:
In a setup with G7 -> xDrip (current stable) -> AAPS a new sensor (the second after migrating from G6) was inserted and -following old habits- started about 3 hours later. This resulted in the Sensor being reported as "Sensor failed 7".
Half a day later, after returning home, a new sensor was inserted to replace the failed one, and started immediately using xDrip. This sensor kept showing the same "Sensor failed 7" message.
22 hours later, it did show some - off - data in the below 50mg/dl range for an hour, before going silent again. 
Metered readings were around 90mg/dl. No calibration was attempted, and later in the day a loaner Dexcom receiver was obtained and supplied with the code. It picked up the connection immediately, displayed the last three hours of data, and is running happily since then with surprisingly precise readings.

With some advice, xDrip was switched to the nightly path and updated, the settings scanned from the instructions and all settings verified.
After connecting to the G7, it went directly to showing "Sensor failed 7". The receiver had been stored away in the microwave as not to interfere meanwhile, but after being brought back it picked up service without a glitch.

Any advice to get things running? The DiaKEM app is not an option at the moment, since the phone (Samsung) is not rooted (The Samsung-specific notes in the settings/instructions were observed).

AAPS is quite unhappy without readings, as is the wearer of all those contraptions.

Thanks in advance

**Comment by Navid200** [comment]
There is no interference between the Dexcom receiver and xDrip.
There will be interference between the Dexcom app and xDrip.

Please post a screenshot of this status page: https://navid200.github.io/xDrip/docs/StatusG5G6.html

**Comment by th122** [comment]
Meanwhile I received the screenshots from the wearer of that combination, I asked for status screen and log: 
![xdrip-chfail-status](https://github.com/NightscoutFoundation/xDrip/assets/17241496/ec3d195a-b4b6-45b0-8721-98eb8d6c00ff)
![xdrip-chfail-log](https://github.com/NightscoutFoundation/xDrip/assets/17241496/c4e364ba-2c55-4c8d-874a-1c17b9ce1864)

(I do have status and log screens from the 2023-11-14 versions as well, but I guess they're of less interest).

Thanks a lot - I'm somewhat out of ideas other than a complete re-setup from scratch, which I'd rather avoid.

**Comment by th122** [comment]
Yeesh, I should have looked at these screenshots at display resolution. The Dexcom app isn't installed on that phone, I know they're on the same channel. No own experience with G7 though, and the community member wearing the thing has just recently been switched to it. First one went fine, second one broke things (may be due to user error/too late activating the thing).

**Comment by jamorham** [comment]
It is possible that I am interpreting that status code as a failure when it isn't which would explain the differences. Ideally if this situation is still occurring, enable ob1 debugging and collect some data and then use the log upload feature. This would allow me to see exactly what is happening and make any required adjustments. Thanks. https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector

**Comment by th122** [comment]
We just uploaded a second set of logfiles (first one was two days ago), since this is the last we are able to fetch with this sensor (G7, Rev 05).
The session ended arount 17oo local time. Once more, we removed all BT pairings from the phone, and restarted the collector. It did reconnect to the sensor, and, after getting a reading, kept on displaying the mysterious "failed 7" on the status page. 
We're now goint to set a new G7 from the same batch, please keep your fingers crossed...

**Reply by Navid200** [reply]
There is no reason to use multiple sensors.
Do you not have a phone that can use the Dexcom app?

**Reply by Navid200** [reply]
I hope what you mean is that you are using each sensor with the receiver.  And you are hoping to get it work with xDrip also.

What is the Samsung model?
Do you mind editing the very first post and in the very first line including the phone brand and model so that one can get that info right away without having to read too many lines?

**Reply by Navid200** [reply]
Are you sure you have disabled battery optimization?
On Samsung, that may be under device care.

**Comment by th122** [comment]
My friend doesn't want to use the Dexcom app, since she wouldn't want her data to be sent to Descom and all associated data-grabbers. Her phone is not rooted, so there's no firewall. She has been happy with xDrip since she started looping some years ago (G5, G6, and since she's allergic to the glue, G7 now). 

The new sensor just started, that one looks fine (values are still off, compared to the old sensor that's still connected to the receiver. For bonding, we removed all BT pairings but the pump. To avoid scanning for the removed sensor, we put in the transmitter ID for a no longer existing G6, then switched to the new G7. Got a pairing request then for both the old G7 and the new subsequently, removed the pairing for the old one from the BT settings, and within a few minutes xDrip - eventually - got rid of the sensor "failed 7" message, and displayed warming up. It's showing it's first readings now, they're quite a bit off, (74, rising to 107 wthin a few readings, compared to fingerstick 141, receiver following the old sensor 139). 
Looks a lot better, I would love to learn what went wrong with the other sensor, though.
Can send the log from the startup of this sensor as well.

**Comment by th122** [comment]
Info in first post updated, Battery optimization is off for all applications. Log from startup just got sent.

**Reply by Navid200** [reply]
Is the phone updated to the latest available Android 11 update from Samsung?
Please note that the readings you are getting in xDrip are exactly the readings coming from G7.  xDrip makes no change to them.

I'm not sure why you refer to the phone being rooted or not and why you refer to apps like Diakem.

There are many people using G7 with xDrip.  We use phones that are not rooted.  There are even some who use Samsung!  
I have been using G7 with xDrip for a long time.  I am on my twelfth or thirteenth G7 now.  I use Pixel.  It does not disconnect.

**Comment by th122** [comment]
I'm not using a Samsung either for myself, but that's what my friend has and prefers (even though she doesn't use the loop phone for telephone services). Diakem would have been an attempt to get readings from the sensor that didn't cooperate so well with xDrip, since the (borrowed) receiver did connect without problems, 
Disconnects never seemed to be a problem with her Samsung (and it's on Android 11, Security patches as of May 2023)
My own phone (Pocophone F1 currently) runs very reliably without disconnect, but I'm still using G6. (would prefer G5, but that ship has sailed, unfortunately). 
Why rooted - I prefer being able to access the stored data of programs, do backups, and keep them from communicating further than they're allowed to...

**Comment by Navid200** [comment]
After (and including) the 2024.08.07 xDrip [release](https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.08.07), sensor failed 7 condition will provide readings if readings are available.
